File: Set1_1000/1209_Remove_All_Adjacent_Duplicates_In_String_II.java
class Pair {
    char ch;
    int count;
    
    Pair(char ch, int count) {
        this.ch = ch;
        this.count = count;
    }
}

class Solution {
    public String removeDuplicates(String s, int k) {
        StringBuilder sb = new StringBuilder();
        Stack<Pair> stack = new Stack<>();
        for (char c : s.toCharArray()) {
            if (!stack.isEmpty() && stack.peek().ch == c) {
                stack.peek().count += 1;
            }
            else {
                stack.push(new Pair(c, 1));
            }
            if (stack.peek().count == k) {
                stack.pop();
            }
        }
        while (!stack.isEmpty()) {
            Pair current = stack.pop();
            for (int i=0; i<current.count; i++)
                sb.append(current.ch);
        }
        return sb.reverse().toString();
    }
}

File: Set1_1000/1209_Remove_All_Adjacent_Duplicates_In_StringII.py
class Solution:
    def removeDuplicates(self, s: str, k: int) -> str:
        size = len(s)
        count = 1
        for i in range(1, size):
            if s[i-1] == s[i]:
                count += 1
            elif s[i-1] != s[i]:
                count = 1
            if count == k:
                s = s.replace(s[i-k+1:i+1], '')
                return self.removeDuplicates(s, k)
        return s
